### PR TITLE
part 6 was implemented

### PR DESCRIPTION
**Pull request by Mandybur800**

### CODE REVIEW

File: src/main/java/com/manufacturing/controllers/car/AddDriverToCarController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        req.getRequestDispatcher("/WEB-INF/views/car/addDriver.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Long driverId = Long.valueOf(req.getParameter("driver_id"));
+        Long carId = Long.valueOf(req.getParameter("car_id"));
+        Car car = carService.get(carId);
+        Driver driver = driverService.get(driverId);
```
Comment (Vedroid) [inline src/main/java/com/manufacturing/controllers/car/AddDriverToCarController.java]: What if the specified id is not in the database?

File: src/main/java/com/manufacturing/controllers/car/CreateCarController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String model = req.getParameter("model");
+        String manufacturer = req.getParameter("manufacturer");
+        Long manufacturerId = Long.parseLong(manufacturer);
+        Car car = new Car(model, manufacturerService.get(manufacturerId));
```
Comment (Vedroid) [inline src/main/java/com/manufacturing/controllers/car/CreateCarController.java]: What if the specified id is not in the database

File: src/main/webapp/web.xml [hunk 3]
```diff
@@ -7,9 +7,99 @@
...
+        <url-pattern>/car/create</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>com.manufacturing.controllers.car.AddDriverToCarController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/car/addDriver</url-pattern>
```
Comment (Vedroid) [inline src/main/webapp/web.xml]: Bad URLs naming
[checklist](https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/08_servlets_practice_checklist.html)

**Review by Vedroid — CHANGES_REQUESTED** [review]

**Review by Vedroid — APPROVED** [review]

File: src/main/java/com/manufacturing/controllers/car/AddDriverToCarController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            throws IOException, ServletException {
+        try {
+            Long driverId = Long.valueOf(req.getParameter("driver_id"));
+            Long carId = Long.valueOf(req.getParameter("car_id"));
+            Car car = carService.get(carId);
+            Driver driver = driverService.get(driverId);
+            carService.addDriverToCar(driver,car);
+            resp.sendRedirect(req.getContextPath() + "/");
+        } catch (NoSuchElementException e) {
+            req.getRequestDispatcher("/WEB-INF/views/incorrect.jsp").forward(req, resp);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/manufacturing/controllers/car/AddDriverToCarController.java:39]: you may add an exception as an attribute and show it in the incorrect page

File: src/main/java/com/manufacturing/controllers/car/CreateCarController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException, ServletException {
+        try {
+            String model = req.getParameter("model");
+            String manufacturer = req.getParameter("manufacturer");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/manufacturing/controllers/car/CreateCarController.java]:
```suggestion
            Long manufacturerId = Long.parseLong(req.getParameter("manufacturer_id"));
```

File: src/main/java/com/manufacturing/controllers/car/CreateCarController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            Long manufacturerId = Long.parseLong(manufacturer);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/manufacturing/controllers/car/CreateCarController.java]:
```suggestion
```

File: src/main/java/com/manufacturing/controllers/car/CreateCarController.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            Car car = new Car(model, manufacturerService.get(manufacturerId));
+            carService.create(car);
+            resp.sendRedirect(req.getContextPath() + "/");
+        } catch (NoSuchElementException e) {
+            req.getRequestDispatcher("/WEB-INF/views/incorrect.jsp").forward(req, resp);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/manufacturing/controllers/car/CreateCarController.java:38]: same

File: src/main/webapp/WEB-INF/views/driver/create.jsp [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+<html>
+<head>
+    <title>Add driver</title>
+</head>
+<body>
+<h1> Please, put information about your driver</h1>
+<form method="post" action="${pageContext.request.contextPath}/driver/create">
+    Please, enter name of driver<input type="text" name="name" required>
+    Please, enter driving license<input type="number" name="license" required>
+    <button type="submit">Create</button>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/driver/create.jsp]:
```suggestion
<form method="post" action="${pageContext.request.contextPath}/driver/create">
    Please, enter name of driver
    <input type="text" name="name" required>
    Please, enter driving license
    <input type="number" name="license" required>
    <button type="submit">Create</button>
```

File: src/main/webapp/WEB-INF/views/driver/create.jsp [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+</form>
+</body>
+</html>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/driver/create.jsp]: add tabulation to your jsp files

File: src/main/webapp/WEB-INF/views/manufacturer/create.jsp [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Add manufacturer</title>
+</head>
+<body>
+<h1> Please, put information about your manufacturer</h1>
+<form method="post" action="${pageContext.request.contextPath}/manufacturer/create">
+    Please, enter name of manufacturer<input type="text" name="name" required>
+    Please, enter country of manufacturer<input type="text" name="country" required>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/manufacturer/create.jsp]:
```suggestion
    Please, enter name of manufacturer
    <input type="text" name="name" required>
    Please, enter country of manufacturer
    <input type="text" name="country" required>
```

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -7,9 +7,99 @@
...
+        <url-pattern>/</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allManufacturers</servlet-name>
+        <servlet-class>com.manufacturing.controllers.manufacturer.GetAllManufacturersController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allManufacturers</servlet-name>
+        <url-pattern>/manufacturer/</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/</url-pattern>
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allDrivers</servlet-name>
+        <servlet-class>com.manufacturing.controllers.driver.GetAllDriversController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allDrivers</servlet-name>
+        <url-pattern>/driver/</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allCars</servlet-name>
+        <servlet-class>com.manufacturing.controllers.car.GetAllCarsController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allCars</servlet-name>
+        <url-pattern>/car/</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteManufacturer</servlet-name>
+        <servlet-class>com.manufacturing.controllers.manufacturer.DeleteManufacturerController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>deleteManufacturer</servlet-name>
+        <url-pattern>/manufacturer/delete</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteDriver</servlet-name>
+        <servlet-class>com.manufacturing.controllers.driver.DeleteDriverController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>deleteDriver</servlet-name>
+        <url-pattern>/driver/delete</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -7,9 +7,99 @@
...
     </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteCar</servlet-name>
+        <servlet-class>com.manufacturing.controllers.car.DeleteCarController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>deleteCar</servlet-name>
+        <url-pattern>/car/delete</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createManufacturer</servlet-name>
+        <servlet-class>com.manufacturing.controllers.manufacturer.CreateManufacturerController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>createManufacturer</servlet-name>
+        <url-pattern>/manufacturer/create</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/manufacturers/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createDriver</servlet-name>
+        <servlet-class>com.manufacturing.controllers.driver.CreateDriverController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>createDriver</servlet-name>
+        <url-pattern>/driver/create</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 19]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createCar</servlet-name>
+        <servlet-class>com.manufacturing.controllers.car.CreateCarController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>createCar</servlet-name>
+        <url-pattern>/car/create</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/create</url-pattern>
```

File: src/main/webapp/web.xml [hunk 20]
```diff
@@ -7,9 +7,99 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>com.manufacturing.controllers.car.AddDriverToCarController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/car/drivers/add</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/drivers/add</url-pattern>
```

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job! Add tabulations to tag in jsp files where needed

**Pull request merged**